apis/core/lib/storage/s3.ts

```diff
@@ -0,0 +1,130 @@
+import aws from 'aws-sdk'
+import { Readable } from 'stream'
+import { PromiseResult } from 'aws-sdk/lib/request'
+import env from '../env'
+import { log } from '../log'
+
+const logError = log.extend('s3').extend('error')
+
+const s3 = new aws.S3({
+  endpoint: env.AWS_S3_ENDPOINT,
+  s3ForcePathStyle: true,
+})
+
+const defaultS3Options = {
+  Bucket: String(env.AWS_S3_BUCKET),
+}
+
+export async function saveFile(path: string, contents: Readable | string): Promise<aws.S3.ManagedUpload.SendData> {
+  const upload = s3.upload({
+    ...defaultS3Options,
+    Body: contents,
+    Key: path,
+  })
+
+  return upload.promise()
+}
+
+export async function deleteFile(path: string): Promise<PromiseResult<aws.S3.DeleteObjectOutput, aws.AWSError>> {
+  return s3.deleteObject({
+    ...defaultS3Options,
+    Key: path,
+  }).promise()
+}
+
+export async function loadFileString(path: string): Promise<string | null> {
+  const file = await s3.getObject({
+    ...defaultS3Options,
+    Key: path,
+  }).promise()
+
+  if (file.Body) {
+    return file.Body.toString()
+  }
+  logError('Could not read file "%s" from S3.', path)
+  return null
+}
+
+export async function loadFileHeadString(path: string, uniqueLines = 20): Promise<string> {
```

Comment:
@vhf Do you think it is possible to extract the business logic and separate it from the S3 code